libraries/networking/src/udt/NetworkSocket.cpp

```diff
@@ -0,0 +1,281 @@
+    switch (socketType) {
+    case SocketType::UDP:
+        _udpSocket.setSocketOption(option, value);
+        break;
+#if defined(WEBRTC_DATA_CHANNELS)
+    case SocketType::WebRTC:
+        _webrtcSocket.setSocketOption(option, value);
+        break;
+#endif
+    default:
+        qCCritical(networking) << "Socket type not specified in setSocketOption()";
+    }
+}
+
+QVariant NetworkSocket::socketOption(SocketType socketType, QAbstractSocket::SocketOption option) {
+    switch (socketType) {
+    case SocketType::UDP:
+        return _udpSocket.socketOption(option);
+#if defined(WEBRTC_DATA_CHANNELS)
+    case SocketType::WebRTC:
+        return _webrtcSocket.socketOption(option);
+#endif
+    default:
+        qCCritical(networking) << "Socket type not specified in socketOption()";
+        return "";
+    }
+}
+
+
+void NetworkSocket::bind(SocketType socketType, const QHostAddress& address, quint16 port) {
+    switch (socketType) {
+    case SocketType::UDP:
+        _udpSocket.bind(address, port);
+        break;
+#if defined(WEBRTC_DATA_CHANNELS)
+    case SocketType::WebRTC:
+        _webrtcSocket.bind(address, port);
+        break;
+#endif
+    default:
+        qCCritical(networking) << "Socket type not specified in bind()";
+    }
+}
+
+void NetworkSocket::abort(SocketType socketType) {
+    switch (socketType) {
+    case SocketType::UDP:
+        _udpSocket.abort();
+        break;
+#if defined(WEBRTC_DATA_CHANNELS)
+    case SocketType::WebRTC:
+        _webrtcSocket.abort();
+        break;
+#endif
+    default:
+        qCCritical(networking) << "Socket type not specified in abort()";
+    }
+}
+
+
+quint16 NetworkSocket::localPort(SocketType socketType) const {
+    switch (socketType) {
+    case SocketType::UDP:
+        return _udpSocket.localPort();
+#if defined(WEBRTC_DATA_CHANNELS)
+    case SocketType::WebRTC:
+        return _webrtcSocket.localPort();
+#endif
+    default:
+        qCCritical(networking) << "Socket type not specified in localPort()";
+        return 0;
+    }
+}
+
+qintptr NetworkSocket::socketDescriptor(SocketType socketType) const {
+    switch (socketType) {
+    case SocketType::UDP:
+        return _udpSocket.socketDescriptor();
+#if defined(WEBRTC_DATA_CHANNELS)
+    case SocketType::WebRTC:
+        return _webrtcSocket.socketDescriptor();
+        return 0;
+#endif
+    default:
+        qCCritical(networking) << "Socket type not specified in socketDescriptor()";
+        return 0;
+    }
+}
+
+
+qint64 NetworkSocket::writeDatagram(const QByteArray& datagram, const HifiSockAddr& sockAddr) {
+    switch (sockAddr.getSocketType()) {
+    case SocketType::UDP:
+        // WEBRTC TODO: The Qt documentation says that the following call shouldn't be used if the UDP socket is connected!!!
+        // https://doc.qt.io/qt-5/qudpsocket.html#writeDatagram
+        return _udpSocket.writeDatagram(datagram, sockAddr.getAddress(), sockAddr.getPort());
```

Comment:
Why use `writeDatagram` anyway, against the recommendation, then? It seems easy to call `write` here as recommended.

Reply:
This is the way it's currently done in `master` for UDP packets and should be addressed there rather than here.
https://github.com/vircadia/vircadia/issues/1279